raiden/api/rest.py

```diff
@@ -984,24 +984,25 @@ def initiate_payment(
         except InsufficientFunds as e:
             return api_error(errors=str(e), status_code=HTTPStatus.PAYMENT_REQUIRED)
 
-        if payment_status.payment_done.get() is False:
+        result = payment_status.payment_done.get()
+
+        if isinstance(result, EventPaymentSentFailed):
             return api_error(
                 errors="Payment couldn't be completed "
-                "(insufficient funds, no route to target or target offline).",
+                "(insufficient funds, no route to target or target offline). " + result.reason,
```

Comment:
You could change the three lines above to:
```
return api_error(
    errors=f"Payment couldn't be completed because: {result.reason}",

```

The message before the reason is not necessary anymore because in the places we emit the `EventPaymentSentFailed` and we provide the reason.